File: 1306-minimum-absolute-difference/minimum-absolute-difference.java
class Solution {
    public List<List<Integer>> minimumAbsDifference(int[] arr) {

        List<List<Integer>> new_list = new ArrayList<>();

        Arrays.sort(arr);

        int minAbsDiff = Integer.MAX_VALUE;
        for (int i = 1; i <= arr.length-1; i++) {
            minAbsDiff = Math.min(minAbsDiff, arr[i] - arr[i - 1]);
        }

        for (int i = 1; i <= arr.length-1; i++) {
            if (arr[i] - arr[i - 1] == minAbsDiff) {
                new_list.add(Arrays.asList(arr[i - 1], arr[i]));
            }
        }

        return new_list;
    }
}

File: 1306-minimum-absolute-difference/minimum-absolute-difference.py
class Solution:
    def minimumAbsDifference(self, arr: List[int]) -> List[List[int]]:
        new_list = []
        arr.sort()

        minAbsDiff = float('inf')

        for i in range(1, len(arr)):
            minAbsDiff = min(minAbsDiff, arr[i] - arr[i-1])

        for i in range(1, len(arr)):
            if arr[i] - arr[i-1] == minAbsDiff:
                new_list.append([arr[i-1], arr[i]])

        return new_list
